### ISSUE TITLE
"unable to resolve reference to ... for \ref command" in file comment - but it's right there

### ISSUE THREAD

**Issue by eyalroz** [doxygen/doxygen#10719]
**Describe the bug**
I'm doxygen'ing a C++ header file with a namespaced class, `cuda::version_t`. I'm also `@ref`'ing that class in the file comment. Unfortunately - the `'@ref` fails with error:
```
warning: unable to resolve reference to 'cuda::version_t' for \ref command
```
I don't understand why that is: The class is visible and documented.

**To Reproduce**
See attachment. 
[10719.zip](https://github.com/doxygen/doxygen/files/14536258/10719.zip)

**Expected behavior**
No warnings or errors.

**Version**
1.11.0 (42d0f98dcbb309dcae08d1495786add326aaeb41*)

**Comment by albert-github** [comment]
The command `@nocond` doxes not exist, the command is called `@endcond`.

(Note: due to the moment of resolving the `@cond` / `@endcond` it is not possible to redefine the command `nocond` to `@endcond` by means of an `ALIASES`) 

I think the problem is usage.

**Comment by eyalroz** [comment]
@albert-github : Oh yes, you're right... however - it would be helpful to users if they could figure out that this is the actual problem. But - the only warning one gets is about a missing reference. Perhaps if you could emit an error message for an unterminated @nocond ? Or a nested @nocond? Or a @nocond which exceeds scope boundaries (starts inside a class and ends outside of it)?

**Comment by albert-github** [comment]
I thought the message would be present as in the code I see:
```
     warn(ctx->fileName,ctx->lineNr,"Conditional section%sdoes not have "
         "a corresponding \\endcond command within this file.",qPrint(sectionInfo));
```

further inspecting the `version.hpp` and searching for:
```
        ///@cond
        ///@nocond
        ///@cond
        ///@nocond
///@cond
///@endcond
```

normally `@cond` parts can be nested and when replacing all the `@cond` with `@cond A` we get the message, it looks like that the message is not given when we have nested "anonymous" `cond` commands.
Some further investigation is needed ...

**Comment by albert-github** [comment]
've just pushed a proposed patch, pull request #10720

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
